File: src/main/java/io/felipeandrade/gsw2/material/vanilla/StoneMaterial.java
package io.felipeandrade.gsw2.material.vanilla;

import net.minecraft.item.Item;
import net.minecraft.item.ToolMaterials;

import java.util.Collections;
import java.util.List;

import io.felipeandrade.gsw2.GSWItemGroup;
import io.felipeandrade.gsw2.common.GSWColor;
import io.felipeandrade.gsw2.item.GSWItem;
import io.felipeandrade.gsw2.item.tool.GSWHammer;
import io.felipeandrade.gsw2.item.tool.GSWTool;
import io.felipeandrade.gsw2.material.ColorProviderMaterialItem;
import io.felipeandrade.gsw2.material.GSWMaterial;
import io.felipeandrade.gsw2.material.GSWMaterialItem;

public class StoneMaterial extends GSWMaterial {

    public static final GSWMaterial MATERIAL = new StoneMaterial();

    public StoneMaterial() {
        super("stone", GSWColor.gray_grey);
    }

    @Override
    public List<? extends GSWTool> allTools() {
        return Collections.singletonList(HAMMER);
    }

    @Override
    public List<? extends GSWItem> allItems() {
        return Collections.singletonList(GEAR);
    }


    public static final GSWMaterialItem GEAR = new ColorProviderMaterialItem("gear", MATERIAL, new Item.Settings().group(GSWItemGroup.MATERIALS));

    public static final GSWTool HAMMER = new GSWHammer(MATERIAL, ToolMaterials.STONE, new Item.Settings().group(GSWItemGroup.TOOLS));

}

File: gsw-mod/src/main/kotlin/io/felipeandrade/gsw/material/vanilla/StoneMaterial.kt
package io.felipeandrade.gsw.material.vanilla

import io.felipeandrade.gsw.item.tool.GSWHammer
import io.felipeandrade.gsw.item.tool.GSWTool
import io.felipeandrade.gsw.material.GSWMaterial
import net.fabricmc.fabric.api.datagen.v1.provider.FabricRecipeProvider
import net.minecraft.data.server.recipe.RecipeExporter
import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraft.item.Items
import net.minecraft.item.ToolMaterials

class StoneMaterial : GSWMaterial("stone") {

    override fun allTools(): List<GSWTool> = listOf(HAMMER)
    override fun generateRecipes(provider: FabricRecipeProvider, exporter: RecipeExporter) {
        HAMMER.offerRecipe(exporter, Items.STONE, Items.STICK)
    }

    companion object {
        val MATERIAL: GSWMaterial = StoneMaterial()
        val HAMMER: GSWHammer = GSWHammer(MATERIAL, ToolMaterials.STONE, Item.Settings())

        fun addMaterialsToItemGroup(entries: ItemGroup.Entries){}
        fun addToolsToItemGroup(entries: ItemGroup.Entries): Unit = with(entries) {
            add(HAMMER)
        }
    }
}